src/app/views.py

```diff
@@ -79,35 +78,70 @@ def submitNewLicense(request):
     """ View for submit new licenses
     returns submit_new_license.html template
     """
-    context_dict={}
-    if request.method == 'POST':
-        form = LicenseRequestForm(request.POST, auto_id='%s')
-        if form.is_valid() and request.is_ajax():
-            licenseName = form.cleaned_data['fullname']
-            licenseIdentifier = form.cleaned_data['shortIdentifier']
-            licenseOsi = form.cleaned_data['osiApproved']
-            licenseSourceUrls = [form.cleaned_data['sourceUrl']]
-            licenseHeader = form.cleaned_data['licenseHeader']
-            licenseNotes = form.cleaned_data['notes']
-            licenseText = form.cleaned_data['text']
-            userEmail = form.cleaned_data['userEmail']
-            xml = generateLicenseXml(licenseOsi, licenseIdentifier, licenseName,
-                licenseSourceUrls, licenseHeader, licenseNotes, licenseText)
-            now = datetime.datetime.now()
-            licenseRequest = LicenseRequest(fullname=licenseName,shortIdentifier=licenseIdentifier,
-                submissionDatetime=now, userEmail=userEmail, xml=xml)
-            licenseRequest.save()
-            statusCode = createIssue(licenseName, licenseIdentifier, licenseSourceUrls, licenseOsi)
-            data = {'statusCode' : str(statusCode)}
-            return JsonResponse(data)
-
+    context_dict = {}
+    ajaxdict = {}
+    if request.method=="POST":
+        if not request.user.is_authenticated():
+            return HttpResponseRedirect(settings.LOGIN_URL)        
```

Comment:
This return is making the ajax behave unexpectedly when the user is not logged in. The correct code would be same as the expect block of UserSocialAuth.DoesNotExist:

```suggestion
            if (request.is_ajax()):
                ajaxdict["type"] = "auth_error"
                ajaxdict["data"] = "Please login using GitHub to use this feature."
                response = dumps(ajaxdict)
                return HttpResponse(response,status=401)
            return HttpResponse("Please login using GitHub to use this feature.",status=401)
```